tests/profile_check.py

```diff
@@ -0,0 +1,480 @@
+                    if fileMsg:
+                        messages.append(fileMsg)
+
+                elif RE_PROFILE_END.search(line):
+                    if getCurrentProfile(nestingStacker):
+                        if not nestingStacker:
+                            messages.append({'filename':   fullpath,
+                                             'profile':    None,
+                                             'severity':   'DEGRADED',
+                                             'line':       n,
+                                             'reason':     "Unbalanced parenthesis?",  # not fully covered
+                                             'suggestion': None})
+                        else:
+                            profileIdentity = '//'.join(nestingStacker)
+                            localExists_eol[profileIdentity] = n
+                            del nestingStacker[-1]  # remove last
+
+                lastLineNum = n
+
+    except PermissionError:
+        exceptionMsg = 'Unable to read the file (PermissionError)'
+
+    except UnicodeDecodeError:
+        exceptionMsg = 'Unable to read the file (UnicodeDecodeError)'
+
+    except FileNotFoundError:
+        exceptionMsg = 'No such file or directory (FileNotFoundError)'
+
+    if exceptionMsg:
+        messages.append({'filename':   fullpath,
+                         'profile':    None,
+                         'severity':   'NOTICE',
+                         'line':       None,
+                         'reason':     exceptionMsg,
+                         'suggestion': None})
+
+    # Ensure proper header is present
+    if len(gotHeaders) < 3:
+        combinedHeader = '\n'.join(headers)
+        messages.append({'filename':   fullpath,
+                         'profile':    None,
+                         'severity':   'WARNING',
+                         'line':       1,
+                         'reason':     'No proper header',
+                         'suggestion': combinedHeader})
+
+    # Ensure ABI is present
+    changeAbi = False
+    abi = 'abi <abi/4.0>,'
+    if gotAbi:
+        if gotAbi.strip() != abi:
+            changeAbi = True
+    else:
+        changeAbi = True
+
+    if changeAbi:
+        messages.append({'filename':   fullpath,
+                         'profile':    None,
+                         'severity':   'WARNING',
+                         'line':       None,
+                         'reason':     'ABI is required',
+                         'suggestion': abi})
+
+    # Ensure singular '@{exec_path}'
+    if not gotAttach:
+        messages.append({'filename':   fullpath,
+                         'profile':    None,
+                         'severity':   'WARNING',
+                         'line':       None,
+                         'reason':     "'@{exec_path}' must be defined as main path attachment",
+                         'suggestion': None})
```

Comment:
Should we ignore orphan profiles like `apparmor.d/groups/children/`? (in future PRs)

Reply:
Yes, they will have to be ignored.

Reply:
`profiles-m-r/pam_roles` move to `profiles-m-r/pam/` and also exclude?